pkg/sql/create_table.go

```diff
@@ -80,12 +80,22 @@ func (p *planner) CreateTable(ctx context.Context, n *tree.CreateTable) (planNod
 		}
 
 		// Synthesize an input column that provides the default value for the
-		// hidden rowid column.
+		// hidden rowid column, if none of the provided columns are specified
+		// as the PRIMARY KEY.
 		synthRowID = true
+		for _, def := range n.Defs {
+			if d, ok := def.(*tree.ColumnTableDef); !ok {
+				return nil, errors.Errorf("failed to cast type to ColumnTableDef\n")
+			} else if d.PrimaryKey {
+				synthRowID = false
+				break
+			}
+		}
 	}
 
 	ct := &createTableNode{n: n, dbDesc: dbDesc, sourcePlan: sourcePlan}
 	ct.run.synthRowID = synthRowID
+	ct.run.fromHeuristicPlanner = true
```

Comment:
can you comment this / where used below (explaining the _why_ in addition to the what)?